rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -2949,7 +3032,45 @@ error_code sys_fs_get_mount_info(ppu_thread&, vm::ptr<CellFsMountInfo> info, u32
 
 error_code sys_fs_mount(ppu_thread&, vm::cptr<char> dev_name, vm::cptr<char> file_system, vm::cptr<char> path, s32 unk1, s32 prot, s32 unk3, vm::cptr<char> str1, u32 str_len)
 {
-	sys_fs.todo("sys_fs_mount(dev_name=%s, file_system=%s, path=%s, unk1=0x%x, prot=0x%x, unk3=0x%x, str1=%s, str_len=%d)", dev_name, file_system, path, unk1, prot, unk3, str1, str_len);
+	sys_fs.warning("sys_fs_mount(dev_name=%s, file_system=%s, path=%s, unk1=0x%x, prot=0x%x, unk3=0x%x, str1=%s, str_len=%d)", dev_name, file_system, path, unk1, prot, unk3, str1, str_len);
+
+	if (!g_ps3_process_info.has_root_perm())
+	{
+		return CELL_ENOSYS;
+	}
+
+	const auto [path_error, vpath] = translate_to_sv(path);
+
+	if (path_error)
+	{
+		return { path_error, vpath };
+	}
+
+	if (vpath != "/dev_hdd1")
+	{
+		return CELL_OK; // We are not supporting mounting devices other than /dev_hdd1 via this syscall currently
+	}
+
+	return vfs::mount(vpath, lv2_fs_object::get_vfs(vpath)) ? CELL_OK : CELL_EIO;
```

Comment:
This is a tricky one, this results in a type of u32 which would be logged as an error when converting to error_code (it's built this way that only specific types are not errors). You xan avoid it by not using s ternary and using simple if or even assert instead.